sorc/chgres_cube.fd/model_grid.F90

```diff
@@ -781,12 +795,334 @@ subroutine define_input_grid_gfs_grib2(localpet, npets)
 
  end subroutine define_input_grid_gfs_grib2
  
+ subroutine define_input_grid_grib2(localpet, npets)
+
+
+ use netcdf
+ use wgrib2api
+ use program_setup, only       : grib2_file_input_grid, data_dir_input_grid, &
+                                  fix_dir_input_grid
+ implicit none
+
+ include 'mpif.h'
+
+ character(len=500)           :: the_file, temp_file
+
+ integer, intent(in)          :: localpet, npets
+
+ integer                      :: error, extra, i, j, clb(2), cub(2) 
+
+ real(esmf_kind_r4), allocatable       :: latitude_one_tile(:,:), lat_corners(:,:)
+ real(esmf_kind_r4), allocatable       :: longitude_one_tile(:,:), lon_corners(:,:)
+ real(esmf_kind_r8)                    :: lat_target(i_target,j_target), &
+                                          lon_target(i_target,j_target)
+ real(esmf_kind_r8)                    :: deltalon, dx
+ integer                               :: ncid,id_var, id_dim
+ real(esmf_kind_r8), pointer           :: lat_src_ptr(:,:), lon_src_ptr(:,:)
+ character(len=10000)            :: cmdline_msg, temp_msg, temp_msg2
+ character(len=10)              :: temp_num = 'NA'
+
+ num_tiles_input_grid = 1
+
+ !inv_file = "chgres.inv"
+ the_file = trim(data_dir_input_grid) // "/" // grib2_file_input_grid
+ temp_file = trim(fix_dir_input_grid)//"/latlon_grid3.32769.nc"
+
+ call ESMF_FieldGather(latitude_target_grid, lat_target, rootPet=0, tile=1, rc=error)
+ if(ESMF_logFoundError(rcToCheck=error,msg=ESMF_LOGERR_PASSTHRU,line=__LINE__,file=__FILE__)) &
+      call error_handler("IN FieldGather", error)
+ call ESMF_FieldGather(longitude_target_grid, lon_target, rootPet=0, tile=1, rc=error)
+ if(ESMF_logFoundError(rcToCheck=error,msg=ESMF_LOGERR_PASSTHRU,line=__LINE__,file=__FILE__)) &
+      call error_handler("IN FieldGather", error)
+
+
+  ! Check for the grid template number to see if wgrib2 can access the lat/lon arrays
+  if (localpet==0) then
+    cmdline_msg = "wgrib2 "//trim(the_file)//" -d 1 -Sec3 &> temp.out"
+    call system(cmdline_msg)
+  endif
+  call MPI_BARRIER(MPI_COMM_WORLD, error)
```

Comment:
I spoke with Wesley E. about how to return the template info in memory.  He said you can call grb2_inq with argument 'grid_desc=' to return it as a text string.  More info is here: https://www.cpc.ncep.noaa.gov/products/wesley/wgrib2/grb2_inq.html.  If that works, you won't have to carry around the wgrib2 executable.

Reply:
The latest push to my fork contains a commit that replaces this use of the wgrib2 executable with code that use grb2_inq instead.

Reply:
Great.  That is less 'clunky'.